app/api/event_tag_endpoint.py

```diff
@@ -0,0 +1,38 @@
+from tracardi.service.storage.drivers.elastic.tag import add_tags, remove_tags, load_all_tags
+from pydantic import BaseModel
+from typing import List
+from fastapi import APIRouter
+from app.config import server
+from tracardi.domain.value_object.bulk_insert_result import BulkInsertResult
+
+
+class TagForm(BaseModel):
+    event_type: str = ""
+    tags: List[str] = []
+
+
+router = APIRouter()
+
+
+@router.post("/event/tag/add", tags=["event"], include_in_schema=server.expose_gui_api, response_model=dict)
+async def add_event_tag_endpoint(tag_form: TagForm):
+    updated, result = await add_tags(
+        event_type=tag_form.event_type,
+        tags=tag_form.tags
+    )
+    return {"ERROR": {"details": result.errors}} if result.errors else {"OK": {"new": 1-updated, "updated": updated}}
```

Comment:
If there is an error there should be HTTPExcepion thrown it will convert to 500 and dict with details. This way you return 200 OK with error information.